aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java

```diff
@@ -191,9 +306,9 @@ private AppSyncRequestFactory() {}
      * @param <T> Type of model being mutated
      * @return Mutation doc
      */
-    private static <T extends Model> String buildMutation(Class<T> modelClass, MutationType mutationType)
-            throws DataStoreException {
-
+    private static <T extends Model> String buildMutation(Class<T> modelClass,
+                                                          boolean includePredicate,
```

Comment:
Instead of boolean flag, can we eight:
1. Pass the predicate itself in right here (even though we'll only do a null check, internally), or,
2. Use an `enum` as below
```
enum PredicateOutput {
    GENERATE,
    OMIT;
}
```

Then, when you call this method, it will look like:
```
buildMutation(clazz, PredicateOutput.OMIT, MutationType.CREATE);
```
And the meaning of the second var will be clear.

Or maybe:
```
enum GeneratePredicate {
    TRUE,
    FALSE
}
```
Which is just a fancy boolean with a special, readable name.